test/v8-to-istanbul.js

```diff
@@ -89,6 +88,49 @@ ${'//'}${'#'} sourceMappingURL=data:application/json;base64,${base64Sourcemap}
       // that means it was bale to access the content via the provided sources object
       v8ToIstanbul.sourceTranspiled.should.not.be.undefined()
     })
+
+    it('should clamp line source column >= 0', async () => {
+      const v8ToIstanbul = new V8ToIstanbul(
+        `file://${require.resolve('./fixtures/scripts/needs-compile.compiled.js')}`,
+        0
+      )
+
+      // read the file and find the first end of line char
+      const fileBody = readFileSync(require.resolve('./fixtures/scripts/needs-compile.compiled.js')).toString()
+      const matchedNewLineChar = fileBody.match(/(?<=\r?\n)/u).index
+      assert(fileBody.substring(matchedNewLineChar - 2, matchedNewLineChar) === '\r\n', 'The test fixture is misconfigured!')
```

Comment:
I would be tempted to split this into two test cases, one for the empty `sources` array that you mentioned, and one for the behavior around `\r\n` creating a bad offset. This would help me better understand the fix 👌